pkg/controller/render/machine_config_to_ignition.go

```diff
@@ -0,0 +1,107 @@
+package render
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+
+	baseutil "github.com/coreos/butane/base/util"
+	"github.com/coreos/ignition/v2/config/util"
+	ign3types "github.com/coreos/ignition/v2/config/v3_2/types"
+	mcSpecv1 "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
+	ctrlcommon "github.com/openshift/machine-config-operator/pkg/controller/common"
+	"github.com/openshift/machine-config-operator/test/helpers"
+
+	"gopkg.in/yaml.v3"
+)
+
+const MCDContentPath = "/etc/machine-config-daemon/mcd_content.json"
```

Comment:
This one definitely deserves a comment.  It seems strongly related to 
`MachineConfigEncapsulatedPath = "/etc/ignition-machine-config-encapsulated.json"`

In fact, any reason not to make them the same?

Reply:
I'm not familiar with how that path is used but it appears to be used to signal completion of firstboot: https://github.com/openshift/machine-config-operator/blob/master/pkg/daemon/daemon.go#L614

Could also potentially deduplicate some code with `appendEncapsulated`

I was keeping everything separate so we don't break anything but can combine if that makes more sense

Reply:
See https://github.com/openshift/machine-config-operator/pull/868 - the commit message there tries to explain this

Anyways though...I think you're right that because this file is involved in special "firstboot" semantics, perhaps we should use a separate one to be safer.